src/mlir/verona-mlir.cc

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "CLI/CLI.hpp"
+#include "ast/cli.h"
+#include "ast/module.h"
+#include "ast/parser.h"
+#include "ast/path.h"
+#include "ast/prec.h"
+#include "ast/ref.h"
+#include "ast/sym.h"
+#include "dialect/VeronaDialect.h"
+#include "generator.h"
+#include "mlir/InitAllDialects.h"
+#include "mlir/InitAllPasses.h"
+
+#include "llvm/Support/InitLLVM.h"
+
+namespace
+{
+  // Source file types, to choose how to parse
+  enum class Source
+  {
+    NONE,
+    VERONA,
+    MLIR
+  };
+
+  // Command line options, with output control
+  struct Opt
+  {
+    std::string grammar;
+    std::string filename;
+    std::string output;
+    bool mlir = false;
+    bool llvm = false;
+  };
+
+  // Parse cmd-line and set defaults
+  Opt parse(int argc, char** argv)
+  {
+    CLI::App app{"Verona MLIR"};
```

Comment:
Is this command-line arg thing significantly better than the LLVM one?  Since we're already parsing LLVM args, I'd prefer to not have two command-line arg parsers...

Reply:
I agree. I'm using it because of how the AST parser used to work (had to pass opt). Now it passes the internal results only, so it's not necessary anymore.

I think this could be a following commit, in order to get the big one out of the way.